File: hana/Algorithm/Algorithm/week01/자료구조/Stack.swift
//
//  Stack.swift
//  Algorithm
//
//  Created by hana on 2023/04/19.
//
///문제:정수를 저장하는 스택 구현 - push x, pop(empty: -1), size, empty(1, 0), top(empty: -1)
///입력: 첫째 줄에 명령의 수 N (1 ≤ N ≤ 10,000), 둘째 줄부터 N개의 줄에는 명령어
///출력: 출력 명령당 한 줄씩 출력
///풀이:
///시간복잡도:
///???:

import Foundation

func solution_10828(_ commands:[String]) -> [Int] {
    var array: [Int] = []
    var printArray: [Int] = []
    
    commands.forEach{
        switch String($0.split(separator: " ")[0]){
        case "push":
            array.append(Int($0.split(separator: " ")[1])!)
        case "pop":
            printArray.append(array.popLast() ?? -1)
        case "size":
            printArray.append(array.count)
        case "empty":
            printArray.append(array.isEmpty ? 1 : 0)
        case "top":
            printArray.append(array.last ?? -1)
        default: break
        }
    }
    
    return printArray
}

let input: Int = Int(readLine()!)!
var commands: [String] = []
for _ in 1...input{
    commands.append(readLine()!)
}

solution_10828(commands).forEach{
    print($0)
}

File: hoon/HoonAlgorithm/src/main/kotlin/boj/week01/stack/Stack.kt
package boj.week01.stack

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

/**
 *
 * no.10828
 * https://www.acmicpc.net/problem/10828
 */


// 시간초과
//fun main() {
//
//    val scanner = Scanner(System.`in`)
//    val inputSize = scanner.nextInt()
//    scanner.nextLine() // 버퍼 지우기
//
//    val inputs = mutableListOf<String>()
//    repeat(inputSize) {
//        val input = scanner.nextLine()
//        inputs.add(input)
//    }
//
//    val stack = ArrayDeque<Int>()
//    inputs.forEach {
//        if (it.contains("push")) {
//            stack.addLast(it.substringAfter(" ").toInt())
//        } else if(it.contains("pop")) {
//            if (stack.isNotEmpty()) {
//                println("${stack.last}")
//                stack.removeLast()
//            } else {
//                println("-1")
//            }
//        } else if (it.contains("size")) {
//            println("${stack.size}")
//        } else if (it.contains("empty")) {
//            if (stack.isEmpty()) {
//                println("1")
//            } else {
//                println("0")
//            }
//        } else if (it.contains("top")) {
//            if (stack.isNotEmpty()) {
//                println("${stack.last}")
//            } else {
//                println("-1")
//            }
//        }
//    }
//}



fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val inputSize = br.readLine().toInt()

    val inputs = mutableListOf<String>()
    repeat(inputSize) {
        inputs.add(br.readLine())
    }

    val stack = LinkedList<Int>()
    val result = StringBuilder()
    inputs.forEach {
        when (it.substringBefore(" ")) {
            "push" -> stack.addLast(it.substringAfter(" ").toInt())
            "pop" -> result.appendLine(if (stack.isEmpty()) "-1" else "${stack.removeLast()}")
            "size" -> result.appendLine("${stack.size}")
            "empty" -> result.appendLine(if (stack.isEmpty()) "1" else "0")
            "top" -> result.appendLine(if (stack.isEmpty()) "-1" else "${stack.last}")
        }
    }
    println(result)
}